suite-native/module-settings/src/components/SettingsSectionItem.tsx

```diff
@@ -1,11 +1,53 @@
-import { ListItem, ListItemProps } from '@suite-native/atoms';
-
-export const SettingsSectionItem = ({ title, subtitle, iconName, onPress }: ListItemProps) => (
-    <ListItem
-        onPress={onPress}
-        subtitle={subtitle}
-        title={title}
-        iconName={iconName}
-        hasRightArrow
-    />
-);
+import { ReactNode } from 'react';
+import { ActivityIndicator, TouchableOpacity, View } from 'react-native';
+
+import { Box } from '@suite-native/atoms';
+import { Icon, IconName } from '@suite-common/icons-deprecated';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+import { SettingsSectionItemIcon } from './SettingsSectionItemIcon';
+import { SettingsSectionItemText } from './SettingsSectionItemText';
+
+export type SettingsSectionItemProps = {
+    iconName?: IconName;
+    title: ReactNode;
+    subtitle?: ReactNode;
+    onPress?: () => void;
+    isLoading?: boolean;
+    testID?: string;
+};
+
+const listItemRightContainerStyle = prepareNativeStyle(utils => ({
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginLeft: utils.spacings.medium,
+}));
+
+export const SettingsSectionItem = ({
+    title,
+    subtitle,
+    iconName,
+    onPress,
+    testID,
+    isLoading,
+}: SettingsSectionItemProps) => {
+    const { applyStyle, utils } = useNativeStyles();
+
+    const isDisabled = !onPress || isLoading;
+
+    return (
+        <TouchableOpacity disabled={isDisabled} onPress={onPress} testID={testID}>
+            <Box flexDirection="row">
+                {iconName && <SettingsSectionItemIcon iconName={iconName} />}
```

Comment:
nit: 
It seems, that iconName is always set, so this can maybe be required param?

Reply:
https://github.com/trezor/trezor-suite/pull/14341/commits/86f647ce514f4d47df7c534495510902e9b80f66